deeplabcut/pose_estimation_tensorflow/predict_supermodel.py

```diff
@@ -0,0 +1,638 @@
+    batchsize=1,
+    TFGPUinference=True,
+    modelprefix="",
+    robust_nframes=False,
+    allow_growth=False,
+    init_weights="",
+    save_frames=False,
+    customized_test_config = '',
+):
+    """
+    Makes prediction based on a super animal model. Note right now we only support single animal video inference
+
+    The index of the trained network is specified by parameters in the config file (in particular the variable 'snapshotindex')
+
+    Output: The labels are stored as MultiIndex Pandas Array, which contains the name of the network, body part name, (x, y) label position \n
+            in pixels, and the likelihood for each frame per body part. These arrays are stored in an efficient Hierarchical Data Format (HDF) \n
+            in the same directory, where the video is stored.
+
+    Parameters
+    ----------
+    videos: list
+        A list of strings containing the full paths to videos for analysis or a path to the directory, where all the videos with same extension are stored.
+
+    superanimal_name: str
+        The name of the superanimal model. We currently only support supertopview and superquadruped
+    scale_list: list
+        A list of int containing the target height of the multi scale test time augmentation. By default it uses the original size. Users are advised to try a wide range of scale list when the super model does not give reasonable results
+
+    videotype: string, optional
+        Checks for the extension of the video in case the input to the video is a directory.\n Only videos with this extension are analyzed. The default is ``.avi``
+
+    destfolder: string, optional
+        Specifies the destination folder for analysis data (default is the path of the video). Note that for subsequent analysis this
+        folder also needs to be passed.
+
+    batchsize: int, default from pose_cfg.yaml
+        Change batch size for inference; if given overwrites value in pose_cfg.yaml
+
+    TFGPUinference: bool, default: True
+        Perform inference on GPU with TensorFlow code. Introduced in "Pretraining boosts out-of-domain robustness for pose estimation" by
+        Alexander Mathis, Mert Yüksekgönül, Byron Rogers, Matthias Bethge, Mackenzie W. Mathis Source: https://arxiv.org/abs/1909.11229
+
+    robust_nframes: bool, optional (default=False)
+        Evaluate a video's number of frames in a robust manner.
+        This option is slower (as the whole video is read frame-by-frame),
+        but does not rely on metadata, hence its robustness against file corruption.
+
+    allow_growth: bool, default false.
+        For some smaller GPUs the memory issues happen. If true, the memory allocator does not pre-allocate the entire specified
+        GPU memory region, instead starting small and growing as needed. See issue: https://forum.image.sc/t/how-to-stop-running-out-of-vram/30551/2
+
+    save_frames: bool, default false.
+        If video adaptation is needed later, this flag should be set as True
```

Comment:
That's not intuitive to me; instead of writing frames to disk, cannot we use the VideoReader to get video frames during video adaptation?

Reply:
Can we do random access with VideoReader for SGD?

Reply:
yes, we could generate random indices and pick frames with reader.set_to_frame()

Reply:
I'll add this later to the video adaptation PR, as just discussed.